app/models/targeting.rb

```diff
@@ -11,6 +11,8 @@ class Targeting < ApplicationRecord
 
   belongs_to :user
   belongs_to :bookmark
+  belongs_to :organization, optional: true
+  belongs_to :location, optional: true
```

Comment:
I know I'm coming a bit late with this, but just now I finally had time to read through this properly.

Why did you choose to (optionally) link targeting to an organization and a location directly instead of using the `Taxonomix` module?

It goes against the pattern established in Foreman where things are not linked directly to orgs & locs, but instead can belong to multiple. Also it lacks expressiveness a bit. If I have any-any context when triggering a job, it does not mean it will consider hosts from really all organizations and locations, because it is still limited by the organizations and locations the user is assigned to and this should probably be somehow recorded.